### PR TITLE
 Check for stencil only image view in identity swizzle case 

### PR DESCRIPTION
**Pull request by js6i**
The buffer binding size looks unused now (?), but the other commit sets it to the specified value.

### CODE REVIEW

File: MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.mm [hunk 1]
```diff
@@ -148,7 +148,11 @@
                 MVKBuffer* buffer = (MVKBuffer*)bufferInfo.buffer;
                 bb.mtlBuffer = buffer->getMTLBuffer();
                 bb.offset = buffer->getMTLBufferOffset() + bufferInfo.offset;
-				bb.size = (uint32_t)buffer->getByteCount();
```
Comment (billhollings) [inline MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.mm:151]: @js6i Is there a use-case where this is actually causing problems?

When it hits Metal, `MVKMTLBufferBinding.size` is only used when setting inline bytes, not when using an actual `VkBuffer/MTLBuffer`, so `VK_WHOLE_SIZE` should have no applicability.

@cdavis5e `MVKMTLBufferBinding.size` was introduced in support of SPIRV `OpArrayLength`. Your thoughts on this?

File: MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.mm [hunk 2]
```diff
@@ -493,7 +497,10 @@
 			if (_mvkBuffer) {
 				bb.mtlBuffer = _mvkBuffer->getMTLBuffer();
 				bb.offset = _mvkBuffer->getMTLBufferOffset() + _buffOffset + bufferDynamicOffset;
-				bb.size = (uint32_t)_mvkBuffer->getByteCount();
```
Comment (billhollings) [inline MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.mm:496]: Ditto.

File: MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.mm (hunk 1 above)
Comment (js6i) [reply MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.mm:151]: Not that I know of. I was confused by this line, but didn't see the value used anywhere. If it's intended to stay for some reason, I'd put a comment there.

Comment (cdavis5e) [reply MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.mm:151]: That is used to set the buffer size passed down to the shader. See MoltenVK/MoltenVK/Commands/MVKCommandEncoderState.mm:585 (or thereabouts). This change appears to be correct.

**Pull request merged**